src/parser/parsers/souls.py

```diff
@@ -0,0 +1,78 @@
+from parser.maps import SOUL_UNLOCK_MAP
+
+
+class SoulUnlockParser:
+    def __init__(self, heroes_data):
+        self.heroes_data = heroes_data
+
+    def run(self):
+        parsed_soul_unlocks = {}
+        hero_key_of_parsed_soul_unlocks = None
+
+        # Parse for each hero
+        for hero_key, hero_data in self.heroes_data.items():
+            if 'm_mapLevelInfo' in hero_data:
+                su_datas = hero_data['m_mapLevelInfo']
+                hero_soul_unlocks = self._parse_hero_soul_unlocks(su_datas)
+
+                # Initialize the result
+                if parsed_soul_unlocks == {}:
+                    parsed_soul_unlocks = hero_soul_unlocks
+                    hero_key_of_parsed_soul_unlocks = hero_key
+                # Confirm that the soul unlocks is the same as the previous hero's soul unlocks
+                else:
+                    if parsed_soul_unlocks != hero_soul_unlocks:
+                        raise ValueError(
+                            'Soul unlocks do not match between heroes'
+                            + f'{hero_key_of_parsed_soul_unlocks} and {hero_key}'
+                        )
```

Comment:
ah very nice, i just had this thought pop up, since they show them per hero in game